staff/fede-gomez/pre-course/chapter2/project/script.js

```diff
@@ -0,0 +1,106 @@
+/*********PROJECT AIRLINES*********/
+
+// This is an example of an array of objects... each position of the array contains one object...
+var flights = [
+{id: 00, to: "New York", from: "Barcelona", cost: 700,scale: false},
+{id: 01, to: "Los Angeles", from: "Madrid", cost: 1100,scale: true},
+{id: 02, to: "Paris", from: "Barcelona", cost: 210,scale: false},
+{id: 03, to: "Roma", from: "Barcelona", cost: 150,scale: false},
+{id: 04, to: "London", from: "Madrid", cost: 200,scale: false},
+{id: 05, to: "Madrid", from: "Barcelona", cost: 90,scale: false},
+{id: 06, to: "Tokyo", from: "Madrid", cost: 1500,scale: true},
+{id: 07, to: "Shangai", from: "Barcelona", cost: 800,scale: true},
+{id: 08, to: "Sydney", from: "Barcelona", cost: 150,scale: true},
+{id: 09, to: "Tel-Aviv", from: "Madrid", cost: 150,scale: false}
+];
+
+//declaramos la funcion que calculara el coste medio de los vuelos
+const averageFlightCost = function (flights) {
+  let totalCost = 0;
+  let numberOfFlights = flights.length;
+  flights.forEach(function(flight){
+    totalCost+=flight.cost;
+  });
+  return totalCost/numberOfFlights;
+};
+
+const lastFlights = function (flights) {
+  let last5 = 5;
+  console.log('Los ultimos 5 vuelos del día son los vuelos con destino a:\n\n')
+  for(id = flights.length - 1; last5>=1; id--){
+    console.log(flights[id].to);
+    last5--;
+  }
+};
+
+const eliminateFlight = function (id) {
+  let index = flights.findIndex(flight =>  flight.id == id);
+  console.log('The flight following flight has been removed: \n\n');
+  console.table(flights[index]);
+  flights.splice(id, 1);
+  //comprobamos la nueva array
+  console.table(flights);
+}
+
+const addFlight = function (id, to, from, cost, scale){
+  flights.push({id, to, from, cost, scale});
+  console.log(flights[id]);
+}
+
+//declaramos la variable para el nombre de usuario
+let name;
+
+//declaramos la funcion principal
+const mainFunction = function () {
+  //Se preguntará por el nombre de usuario y dará la bienvenida.
+  name = prompt('Bienvenido. Cual es tu nombre?');
+  console.log('Bienvenido ' + name + '. Estos son los vuelos programados para hoy:\n\n');
+  //esta variable guardará la cantidad de vuelos con escala:
+  let numberOfFlightsWithScales = 0;
+  //El usuario visualizará todos los vuelos disponibles de una forma amigable:
+  //El vuelo con origen: Barcelona, y destino: Madrid tiene un coste de XXXX€ y no realiza ninguna escala.
+  flights.forEach(function(flight){
+    if(flight.scale === true){
```

Comment:
doing that type of logic operations with booleans is a bad practice, better use only
```javascript
 if(flight.scale){
```

Reply:
Indeed. Thanks!